x/params/store/store.go

```diff
@@ -0,0 +1,171 @@
+package store
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Individual parameter store for each keeper
+type Store struct {
+	cdc  *codec.Codec
+	key  sdk.StoreKey
+	tkey sdk.StoreKey
+
+	space []byte
+}
+
+// NewStore constructs a store with namestore
+func NewStore(cdc *codec.Codec, key sdk.StoreKey, tkey sdk.StoreKey, space string) Store {
+	return Store{
+		cdc:  cdc,
+		key:  key,
+		tkey: tkey,
+
+		space: []byte(space),
+	}
+}
+
+// Returns a KVStore identical with ctx,TransientStore(s.key).Prefix()
+func (s Store) kvStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.KVStore(s.key).Prefix(append(s.space, '/'))
+}
+
+// Returns a KVStore identical with ctx.TransientStore(s.tkey).Prefix()
+func (s Store) transientStore(ctx sdk.Context) sdk.KVStore {
+	return ctx.TransientStore(s.tkey).Prefix(append(s.space, '/'))
+}
+
+// Get parameter from store
+func (s Store) Get(ctx sdk.Context, key string, ptr interface{}) {
+	store := s.kvStore(ctx)
+	bz := store.Get([]byte(key))
+	err := s.cdc.UnmarshalJSON(bz, ptr)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// GetIfExists do not modify ptr if the stored parameter is nil
+func (s Store) GetIfExists(ctx sdk.Context, key string, ptr interface{}) {
+	store := s.kvStore(ctx)
+	bz := store.Get([]byte(key))
+	if bz == nil {
+		return
+	}
+	err := s.cdc.UnmarshalJSON(bz, ptr)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Get raw bytes of parameter from store
+func (s Store) GetRaw(ctx sdk.Context, key string) []byte {
+	store := s.kvStore(ctx)
+	res := store.Get([]byte(key))
+	return res
+}
+
+// Check if the parameter is set in the store
+func (s Store) Has(ctx sdk.Context, key string) bool {
+	store := s.kvStore(ctx)
+	return store.Has([]byte(key))
+}
+
+// Returns true if the parameter is set in the block
+func (s Store) Modified(ctx sdk.Context, key string) bool {
+	tstore := s.transientStore(ctx)
+	return tstore.Has([]byte(key))
+}
+
+// Set parameter, return error if stored parameter has different type from input
+func (s Store) Set(ctx sdk.Context, key string, param interface{}) {
```

Comment:
I think we should add to this that the parameter is set in both the transient and non-transient stores.